src/tx/write/ssi.rs

```diff
@@ -0,0 +1,1097 @@
+    ///
+    /// ```
+    /// # use fjall::{Config, Keyspace, PartitionCreateOptions};
+    /// #
+    /// # let folder = tempfile::tempdir()?;
+    /// # let keyspace = Config::new(folder).open_transactional()?;
+    /// # let partition = keyspace.open_partition("default", PartitionCreateOptions::default())?;
+    /// #
+    /// let mut tx = keyspace.write_tx()?;
+    /// tx.insert(&partition, "a", "abc");
+    /// tx.insert(&partition, "f", "abc");
+    /// tx.insert(&partition, "g", "abc");
+    ///
+    /// assert_eq!(2, tx.range(&partition, "a"..="f").count());
+    /// assert_eq!(0, keyspace.read_tx().range(&partition, "a"..="f").count());
+    /// #
+    /// # Ok::<(), fjall::Error>(())
+    /// ```
+    #[must_use]
+    pub fn range<'b, K: AsRef<[u8]> + 'b, R: RangeBounds<K> + 'b>(
+        &'b mut self,
+        partition: &'b TxPartitionHandle,
+        range: R,
+    ) -> impl DoubleEndedIterator<Item = crate::Result<KvPair>> + 'b {
+        // TODO: Bound::map 1.77
+        let start: Bound<Slice> = match range.start_bound() {
+            Bound::Included(k) => Bound::Included(k.as_ref().into()),
+            Bound::Excluded(k) => Bound::Excluded(k.as_ref().into()),
+            Bound::Unbounded => Bound::Unbounded,
+        };
+        // TODO: Bound::map 1.77
+        let end: Bound<Slice> = match range.end_bound() {
+            Bound::Included(k) => Bound::Included(k.as_ref().into()),
+            Bound::Excluded(k) => Bound::Excluded(k.as_ref().into()),
+            Bound::Unbounded => Bound::Unbounded,
+        };
+        self.cm.mark_range(&partition.inner.name, (start, end));
+
+        self.inner.range(partition, range)
+    }
+
+    /// Iterates over a range of the transaction's state.
+    ///
+    /// Avoid using an empty prefix as it may scan a lot of items (unless limited).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use fjall::{Config, Keyspace, PartitionCreateOptions};
+    /// #
+    /// # let folder = tempfile::tempdir()?;
+    /// # let keyspace = Config::new(folder).open_transactional()?;
+    /// # let partition = keyspace.open_partition("default", PartitionCreateOptions::default())?;
+    /// #
+    /// let mut tx = keyspace.write_tx()?;
+    /// tx.insert(&partition, "a", "abc");
+    /// tx.insert(&partition, "ab", "abc");
+    /// tx.insert(&partition, "abc", "abc");
+    ///
+    /// assert_eq!(2, tx.prefix(&partition, "ab").count());
+    /// assert_eq!(0, keyspace.read_tx().prefix(&partition, "ab").count());
+    /// #
+    /// # Ok::<(), fjall::Error>(())
+    /// ```
+    #[must_use]
+    pub fn prefix<'b, K: AsRef<[u8]> + 'b>(
+        &'b mut self,
+        partition: &'b TxPartitionHandle,
+        prefix: K,
+    ) -> impl DoubleEndedIterator<Item = crate::Result<KvPair>> + 'b {
+        self.cm.mark_range(
+            &partition.inner.name,
+            lsm_tree::range::prefix_to_range(prefix.as_ref()),
+        );
+
+        self.inner.prefix(partition, prefix)
```

Comment:
```suggestion
        self.range(partition, lsm_tree::range::prefix_to_range(prefix.as_ref()))
```
That should work too, right?

Reply:
Yes